resources/codegen/python/openassetio_codegen/__main__.py

```diff
@@ -0,0 +1,178 @@
+#
+#   Copyright 2022 The Foundry Visionmongers Ltd
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+"""
+Entry point for command-line execution of the specification/trait
+codegen tool.
+"""
+
+# pylint: disable=invalid-name
+
+import argparse
+import inspect
+import sys
+import logging
+
+from . import generate
+from . import generators
+
+#
+# Defaults
+#
+
+# Default log level for message logging
+DEFAULT_LOG_LEVEL = logging.getLevelName(logging.WARNING)
+
+#
+## Helpers
+#
+
+
+def _create_argparser():
+    """
+    Returns an argparse parser configured with the openassetio-codegen
+    flags and options.
+    """
+
+    cmdline = argparse.ArgumentParser(
+        prog="openassetio-codegen",
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        description=inspect.cleandoc(
+            """
+            The openassetio-codegen utility generates code that provides
+            strongly-typed views on an openassetio traits data instance. The
+            tool is capable of generating code in a number of languages from the
+            supplied file, containing the simplified declaration of one or more
+            traits or specifications using the OpenAssetIO traits and
+            specification declaration schema.
+
+            By default, code is generated for all supported unless one or more
+            language flags are specified.
+            """
+        ),
+    )
+
+    cmdline.add_argument(
+        "-d",
+        "--dry-run",
+        action="store_true",
+        help="Load and verify the supplied declarations without generating any code",
+    )
+
+    cmdline.add_argument("input", help="YAML file detailing traits and specifications to generate")
+
+    cmdline.add_argument(
+        "-o",
+        "--output-dir",
+        required=True,
+        help="Generate code under the supplied directory, the utility will attempt to created this"
+        " directory if it does not exist",
+    )
+
+    cmdline.add_argument(
+        "--copyright-owner",
```

Comment:
Yeah true indeed. I had thought about just hard-coding this for now, to avoid a half-formed feature. Happy to log this as a "make copyright better" issue for when we have a use case that demands more than this? (The two options we have right now should make it easy to handle privately owned packages, and anything this org's GitHub space will most likely end up with "Copyright the OpenAssetIO Project" after the ASWF migration).

Reply:
We currently force Apache2 though, which privately owned packages might not want?  

Regardless, I'm happy to log this as a follow-up issue.

Reply:
True, at the mo though if you leave off the flag no license is added.

Reply:
Your idea of swapping to SPDX seems the best

Reply:
Switched to SPDX [here](https://github.com/OpenAssetIO/OpenAssetIO/pull/529/commits/f5b57d4baeb6b0825c7b7a8c0f26355cdc653799).